### PR TITLE
ERP: Modifications in registration form

### PR DESCRIPTION
**Pull request by akanshaaa19**
target issue is #3105

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -21,6 +22,10 @@ const App = () => {
   const [authenticated, setAuthenticated] = useState<any>();
   const [drawerOpen, setDrawerOpen] = useState(true);
 
+  axios
```
Comment (kurund) [inline src/App.tsx:25]: This seems wrong.

**Review by kurund — CHANGES_REQUESTED** [review]
@akanshaaa19 

We need better error handling. If someone tries to register and org does not exist the form fails silently. We should at least show some error.

File: src/App.tsx (hunk 1 above)
Comment (akanshaaa19) [reply src/App.tsx:25]: sorry, this was a mistake

**Comment by akanshaaa19** [comment]
> @akanshaaa19
> 
> 1. We need better error handling. If someone tries to register and org does not exist the form fails silently. We should at least show some helpful message.
> 2. Can you also fix failing unit tests?
> 3. I am also trying to register a new org: `Example NGO` and the backend is failing with an error. Note that `Example NGO` has been added to Frappe.
> 
> Let's discuss.

1. There is error handling for when org doesn't exist in erp.
image
2. done
3. yes amisha is looking into that

**Comment by kurund** [comment]
@akanshaaa19 

Can you update the message to: "Example organization not found". Please contact the Glific support team.

**Comment by akanshaaa19** [comment]
> @akanshaaa19
> 
> Can you update the message to: "Example organization not found". Please contact the Glific support team.

sure @AmishaBisht will be doing that from the BE

**Comment by AmishaBisht** [comment]
> > @akanshaaa19
> > Can you update the message to: "Example organization not found". Please contact the Glific support team.
> 
> sure, @AmishaBisht will be doing that from the BE @kurund

done @akanshaaa19 @kurund

**Comment by kurund** [comment]
@akanshaaa19 

1. In my opinion, we should change the below validation. 

Screenshot 2024-11-17 at 19 41 15

Why is this necessary? If it's something to do with the backend then we can automatically do such conversion.  Let's chat.

2. Error on GST

Screenshot 2024-11-17 at 19 44 44

This just shows red without any message.

3. Incorrect field label

Screenshot 2024-11-17 at 19 46 49

**Comment by kurund** [comment]
@AmishaBisht 

> > > @akanshaaa19
> > > Can you update the message to: "Example organization not found". Please contact the Glific support team.
> > 
> > 
> > sure, @AmishaBisht will be doing that from the BE @kurund
> 
> done @akanshaaa19 @kurund

Looks like it's not done, we still have `Customer`


Screenshot 2024-11-17 at 19 59 35

**Comment by AmishaBisht** [comment]
> @AmishaBisht
> 
> > > > @akanshaaa19
> > > > Can you update the message to: "Example organization not found". Please contact the Glific support team.
> > > 
> > > 
> > > sure, @AmishaBisht will be doing that from the BE @kurund
> > 
> > 
> > done @akanshaaa19 @kurund
> 
> Looks like it's not done, we still have `Customer`
> 
> Screenshot 2024-11-17 at 19 59 35

@kurund the error message I'm receiving is directly from ERP, and I'm passing it along as-is.

**Review by kurund — APPROVED** [review]
Looks good.

Please note the dependency on https://github.com/glific/glific/pull/3893

**Pull request merged**